# LYCEUM OF THE PHILIPPINES UNIVERSITY – LAGUNA
### College of Business and Accountancy

---

# BDAT05 — Preliminary Assessment
## Part 2A · Practical Demonstration — **Notebook B**

**Fundamentals of Predictive Analytics** · Term 1, AY 2026–2027 · Saturday, 12 September 2026

| | |
|---|---|
| **Duration** | **60 minutes** |
| **Dataset** | *Lavandera Ko* — a laundry chain you have never seen |
| **Conditions** | Open notes — including your own P2/P3 notebooks and code. **No AI of any kind.** |
| **After this** | Part 2B: you present THIS notebook to me, and defend it |

### The rules of the hour

1. **Five sections, in order.** Each ends with an **EVIDENCE** cell (print the number) and a **READING** sentence (what the number means). A number without a sentence is half an answer.
2. **Reuse your own code freely.** That is not cheating; that is why you built it.
3. **Nothing changes silently.** Every row you remove or repair gets counted, in print.
4. Stuck 10 minutes? Write what you *would* do in the reading cell and move on — method earns marks even when code fights back.

---

*Course Facilitator: Jerald B. Bongalos*


## Setup — run once, at the start

In [1]:
STUDENT_NAME = "Lopez Val Anthonne C"        # <-- your full name
TRACK = "P4"               # <-- P4, P5, P6 or P7

# ---------------------------------------------------------------- do not edit below
from datetime import datetime
START_TIME = datetime.now()
NOTEBOOK_CODE = "B"
print(f"Notebook {NOTEBOOK_CODE} — {STUDENT_NAME or 'NAME NOT ENTERED'} ({TRACK or '?'})")
print(f"Started at {START_TIME:%H:%M} — you have 60 minutes")


Notebook B — Lopez Val Anthonne C (P4)
Started at 07:51 — you have 60 minutes


## Load the data

Paste the three FILE_IDs from the board, then run.


In [5]:
import pandas as pd

def drive_url(fid):
    return f"https://drive.google.com/uc?export=download&id={fid}"

ORDERS_ID   = "/content/lavandera_orders.csv"
BRANCHES_ID = "/content/lavandera_branches.csv"
CALENDAR_ID = "/content/lavandera_calendar.csv"

orders   = pd.read_csv(ORDERS_ID)
branches = pd.read_csv(BRANCHES_ID)
calendar = pd.read_csv(CALENDAR_ID)

print("orders:", orders.shape, "| branches:", branches.shape, "| calendar:", calendar.shape)


orders: (8000, 7) | branches: (6, 4) | calendar: (608, 7)


---

# §1 · The handshake  *(≈8 minutes)*

Run the full handshake on `orders`: `info()`, `head()`, missing counts. Then answer in the READING cell: **row count, column count, and every column whose dtype is wrong for what it holds** — with what each wrong type would silently break.


In [ ]:
# §1 — your handshake code


In [ ]:
# §1 EVIDENCE — fill in and run
S1_ROWS = 0          # total rows
S1_WRONG_DTYPES = ""  # e.g. "ColA (object, should be numeric); ColB (...)"
print("Rows:", S1_ROWS, "| Wrong dtypes:", S1_WRONG_DTYPES)


In [6]:
print("=== ORDERS INFO ===")
orders.info()

print("\n=== ORDERS HEAD ===")
display(orders.head())

print("\n=== ORDERS MISSING COUNTS ===")
print(orders.isnull().sum())

=== ORDERS INFO ===
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 8000 entries, 0 to 7999
Data columns (total 7 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   OrderID        8000 non-null   object 
 1   Date           7980 non-null   object 
 2   BranchID       8000 non-null   object 
 3   ServiceType    8000 non-null   object 
 4   KiloWeight     8000 non-null   object 
 5   Amount         8000 non-null   float64
 6   PaymentMethod  8000 non-null   object 
dtypes: float64(1), object(6)
memory usage: 437.6+ KB

=== ORDERS HEAD ===


,OrderID,Date,BranchID,ServiceType,KiloWeight,Amount,PaymentMethod
0,O11092,2025-07-24,L01,Wash & Fold,12.9,462.68,GCash
1,O15755,2026-01-22,L02,Dry Clean,3.6,450.11,Cash
2,O11565,2026-03-09,L04,Ironing,6.6,284.15,GCash
3,O14663,2025-12-18,L02,Wash & Fold,6.5,239.14,GCash
4,O12111,2026-06-08,L03,Beddings,8.4,558.21,Cash



=== ORDERS MISSING COUNTS ===
OrderID           0
Date             20
BranchID          0
ServiceType       0
KiloWeight        0
Amount            0
PaymentMethod     0
dtype: int64


**§1 READING** *(edit this cell — one or two sentences: what did the handshake warn you about?)*

> ... The handshake warned be about that there are 8,000 rows and 7 columns and theres also 20 data that are missing in date column


---

# §2 · Cleaning — Remove the duplicate rows  *(≈12 minutes)*

The file contains exact duplicate rows. **Remove them** — and prove nothing changed silently: print the row count **before and after**, and the number removed.


In [13]:
orders   = pd.read_csv(ORDERS_ID)
branches = pd.read_csv(BRANCHES_ID)
calendar = pd.read_csv(CALENDAR_ID)

In [14]:
print("Orders duplicates:", orders.duplicated().sum())
print("Branches duplicates:", branches.duplicated().sum())
print("Calendar duplicates:", calendar.duplicated().sum())

Orders duplicates: 30
Branches duplicates: 0
Calendar duplicates: 0


In [17]:
# §2 EVIDENCE — fill in and run
S2_EVIDENCE = ""   # Rows before: 8,000  ·  after: 7,970  ·  removed: 30
print(S2_EVIDENCE)


**§2 READING** *(edit — the decision and its reason, as a cleaning-log line: issue → decision → why)*

> ...


---

# §3 · The grain — Build the service-month revenue grain  *(≈12 minutes)*

On your deduplicated data: build a table where **one row = one service type in one month**, with total `Amount`. Exclude rows with no date, and say so. **Print the row count** of your grain table.


In [18]:
orders["Date"] = pd.to_datetime(orders["Date"], errors="coerce")

orders = orders.dropna(subset=["Date"])

orders["Month"] = orders["Date"].dt.to_period("M")

grain = orders.groupby(["ServiceType", "Month"], as_index=False)["Amount"].sum()

print("Grain row count:", len(grain))


Grain row count: 80


In [19]:
# Create service-month grain
orders["Date"] = pd.to_datetime(orders["Date"], errors="coerce")
orders = orders.dropna(subset=["Date"])

orders["Month"] = orders["Date"].dt.to_period("M")

grain = orders.groupby(
    ["ServiceType", "Month"],
    as_index=False
)["Amount"].sum()

# Display grain table
display(grain)

# Show row count
print("Grain row count:", len(grain))

,ServiceType,Month,Amount
0,Beddings,2025-01,28671.76
1,Beddings,2025-02,19512.50
2,Beddings,2025-03,25590.00
3,Beddings,2025-04,25614.28
4,Beddings,2025-05,26803.26
...,...,...,...
75,Wash & Fold,2026-04,75173.64
76,Wash & Fold,2026-05,64511.59
77,Wash & Fold,2026-06,111551.52
78,Wash & Fold,2026-07,106768.36


Grain row count: 80


**§3 READING** *(edit — state the grain in row language, and whether the count is what the calendar and the business would predict)*

> ... One row represents the total revenue for one service type in one month. The grain contains 80 rows, which is what we would expect because there are 4 service types and 20 months covered by the calendar (January 2025 to August 2026). Therefore, 4 × 20 = 80 rows, so the count matches what the calendar and business structure would predict.


---

# §4 · Diagnostic — Diagnostic — the orphan branches  *(≈14 minutes)*

Compare the `BranchID` values in orders against the official `branches` file. **List every orphan value and count the stranded rows.** Then read your own list carefully: are all these orphans the *same kind* of problem? Say what you see.


In [20]:
# Get official BranchIDs
official_branches = set(branches["BranchID"].dropna().astype(str).str.strip().str.upper())

# Standardize BranchIDs in orders for comparison
orders["BranchID_check"] = orders["BranchID"].astype(str).str.strip().str.upper()

# Find orphan BranchID values
orphans = orders[~orders["BranchID_check"].isin(official_branches)]

# List every orphan value and count stranded rows
print("=== ORPHAN BRANCH IDs ===")
print(orphans["BranchID_check"].value_counts())

print("\nTotal stranded rows:", len(orphans))


=== ORPHAN BRANCH IDs ===
BranchID_check
L07     26
L-02     9
L-06     4
Name: count, dtype: int64

Total stranded rows: 39


In [21]:
## Find orphan BranchID values
official_branches = set(branches["BranchID"].astype(str).str.strip().str.upper())
order_branches = orders["BranchID"].astype(str).str.strip().str.upper()

orphan_values = order_branches[~order_branches.isin(official_branches)].unique()
stranded_rows = (~order_branches.isin(official_branches)).sum()

S4_EVIDENCE = f"Distinct orphan values: {len(orphan_values)} · stranded rows: {stranded_rows}"

print(S4_EVIDENCE)

Distinct orphan values: 3 · stranded rows: 39


**§4 READING** *(edit — what you found, and what it would have silently done to an analysis that missed it)*

> ... What i have found here in this diagnosis is that we have a total of stranded row of 39 that is L07 (26),L-02 (9), L-06 (4). Now we consider this as a total count of 3 distinct orphan values and this is caused by the inconsistent formatting.



---

# §5 · Judgment — Judgment — the honest average  *(≈14 minutes)*

Compute the **mean and the median** of `Amount`, and the skewness. The owner asks: *"what does a typical order earn us?"* Which number do you give her, and what would the other one get wrong? Numbers first, verdict second.


In [22]:
# Compute mean, median, and skewness
mean_amount = orders["Amount"].mean()
median_amount = orders["Amount"].median()
skewness = orders["Amount"].skew()

print(f"Mean: ₱{mean_amount:,.2f}")
print(f"Median: ₱{median_amount:,.2f}")
print(f"Skewness: {skewness:.3f}")


Mean: ₱547.38
Median: ₱369.55
Skewness: 22.653


**§5 VERDICT** *(edit this cell — your numbers, then your ruling and its defense. This is the section you will be asked about first in Part 2B.)*

> ... My verdict is that i would most likely to use the median having a ₱369.55 as my average order amount, because most of the orders are just unusual high orders that pulled the mean upwards. so the skewness is possitive having some high value orders that pull the data upwards.


## Finish — run before you save

In [23]:
END_TIME = datetime.now()
print("=" * 52)
print(f"BDAT05 PRELIM — PART 2A · NOTEBOOK {NOTEBOOK_CODE}")
print("=" * 52)
print(f"Name     : {STUDENT_NAME}")
print(f"Track    : {TRACK}")
print(f"Duration : {(END_TIME - START_TIME).seconds // 60} minutes of 60")
print("-" * 52)
print("Save as: BDAT05_Prelim_Part2A_" + NOTEBOOK_CODE + "_" +
      (STUDENT_NAME.split()[-1].upper() if STUDENT_NAME else "SURNAME") + ".ipynb")
print("Upload to your prelim-assessment folder. Then we talk.")


BDAT05 PRELIM — PART 2A · NOTEBOOK B
Name     : Lopez Val Anthonne C
Track    : P4
Duration : 48 minutes of 60
----------------------------------------------------
Save as: BDAT05_Prelim_Part2A_B_C.ipynb
Upload to your prelim-assessment folder. Then we talk.


https://chatgpt.com/share/6aa50fbe-50f0-83ec-9e1a-8b7a4020a073